protocols/mdns/src/behaviour.rs

```diff
@@ -279,31 +296,58 @@ impl NetworkBehaviour for Mdns {
         >,
     > {
         while let Poll::Ready(event) = Pin::new(&mut self.if_watch).poll(cx) {
-            let multicast = From::from([224, 0, 0, 251]);
             let socket = self.recv_socket.get_ref();
             match event {
                 Ok(IfEvent::Up(inet)) => {
                     if inet.addr().is_loopback() {
                         continue;
                     }
-                    if let IpAddr::V4(addr) = inet.addr() {
-                        log::trace!("joining multicast on iface {}", addr);
-                        if let Err(err) = socket.join_multicast_v4(&multicast, &addr) {
-                            log::error!("join multicast failed: {}", err);
-                        } else {
-                            self.timeout
-                                .set_interval_at(Instant::now(), self.query_interval);
+                    match self.multicast_addr {
+                        IpAddr::V4(multicast) => {
+                            if let IpAddr::V4(addr) = inet.addr() {
+                                log::trace!("joining multicast on iface {}", addr);
+                                if let Err(err) = socket.join_multicast_v4(&multicast, &addr) {
+                                    log::error!("join multicast failed: {}", err);
+                                } else {
+                                    self.timeout
+                                        .set_interval_at(Instant::now(), self.query_interval);
+                                }
+                            }
+                        }
+                        IpAddr::V6(multicast) => {
+                            if let IpAddr::V6(addr) = inet.addr() {
+                                log::trace!("joining multicast on iface {}", addr);
+                                // TODO: determine interface ID now it generates errors.
```

Comment:
I don't follow. Can you expand on this TODO?

Reply:
not sure, this was written by @majchrzamemil

Reply:
I assume it's related to handling multiple network interfaces for which we have an issue open already, so I'll remove the TODO.